tools/shoestring/tests/commands/test_import_bootstrap.py

```diff
@@ -19,57 +31,86 @@ async def _run_test(is_harvester, is_voter, expected_imports_harvester, expected
 			bootstrap_node_path = Path(bootstrap_directory) / 'nodes/node'
 			bootstrap_node_path.mkdir(parents=True)
 
-			if is_harvester:
+			if enabled_imports.harvester:
 				bootstrap_resources_path = bootstrap_node_path / 'server-config/resources'
 				bootstrap_resources_path.mkdir(parents=True)
 				with open(bootstrap_resources_path / 'config-harvesting.properties', 'wt', encoding='utf8') as outfile:
 					outfile.write('lorem ipsum')
 
-			if is_voter:
+			if enabled_imports.voter:
 				bootstrap_voting_keys_path = bootstrap_node_path / 'votingkeys'
 				bootstrap_voting_keys_path.mkdir(parents=True)
 
+			if enabled_imports.node_key:
+				bootstrap_node_key_path = bootstrap_node_path / 'cert/node.key.pem'
+				bootstrap_node_key_path.mkdir(parents=True)
+
 			# Act:
 			await main([
 				'import-bootstrap',
 				'--config', str(config_filepath),
-				'--bootstrap', str(bootstrap_directory)
+				'--bootstrap', str(bootstrap_directory),
+				*(['--node-key'] if enabled_imports.node_key else [])
 			])
 
 			# Assert:
 			imports = ConfigurationManager(output_directory).lookup(config_filepath.name, [
 				('imports', 'harvester'),
-				('imports', 'voter')
+				('imports', 'voter'),
+				('imports', 'node_key')
 			])
 
 			assert [
-				expected_imports_harvester.format(bootstrap_root=bootstrap_directory),
-				expected_imports_voter.format(bootstrap_root=bootstrap_directory)
+				expected_imports.harvester.format(bootstrap_root=bootstrap_directory),
+				expected_imports.voter.format(bootstrap_root=bootstrap_directory),
+				expected_imports.node_key.format(bootstrap_root=bootstrap_directory)
 			] == imports
 
 
 # pylint: disable=invalid-name
 
 # region success
 
-async def test_can_import_with_neither_harvester_nor_voter():
-	await _run_test(False, False, '', '')
+async def test_can_import_with_neither_harvester_nor_voter_nor_node_key():
+	actual_imports = EnabledImports(False, False, False)
+	expected_imports = build_expected_imports(actual_imports)
+	await _run_test(actual_imports, expected_imports)
+
+
+async def test_can_import_with_harvester_but_not_voter_or_node_key():
+	actual_imports = EnabledImports(True, False, False)
+	expected_imports = build_expected_imports(actual_imports)
+	await _run_test(actual_imports, expected_imports)
+
+
+async def test_can_import_with_voter_but_not_harvester_or_node_key():
+	actual_imports = EnabledImports(False, True, False)
+	expected_imports = build_expected_imports(actual_imports)
+	await _run_test(actual_imports, expected_imports)
 
 
-async def test_can_import_with_harvester_but_not_voter():
-	await _run_test(True, False, '{bootstrap_root}/nodes/node/server-config/resources/config-harvesting.properties', '')
+async def test_can_import_with_node_key_but_not_harvester_or_voter():
+	actual_imports = EnabledImports(False, False, True)
+	expected_imports = build_expected_imports(actual_imports)
+	await _run_test(actual_imports, expected_imports)
```

Comment:
or => nor

Reply:
i think you missed a few 
where you have not ... or it is more correct to have not .. nor